core/commands/resolve.go

```diff
@@ -128,6 +111,23 @@ Resolve the value of an IPFS DAG path:
 			return cmds.EmitOnce(res, &ncmd.ResolvedPath{Path: ipfspath.Path(p.String())})
 		}
 
+		var enc cidenc.Encoder
+		switch {
+		case !cmdenv.CidBaseDefined(req) && !strings.HasPrefix(name, "/ipns/"):
```

Comment:
@petar @Stebalien now that we've introduced `--ipns-base` should we change this behavior to determine the IPNSBase and then do a switch on the ResolvedPath to figure out whether we should use the CID encoder or the IPNS encoder (i.e. if it starts with /ipfs | /ipld or starts with /ipns?)